lang/cs/Org.Apache.REEF.Tests/Functional/Telemetry/MetricsDriver.cs

```diff
@@ -135,11 +139,12 @@ public void OnError(Exception error)
         /// </summary>
         private void UpdateMetrics(TestSystemState systemState)
         {
-            var driverMetrics = new DriverMetrics(EventPrefix + systemState, DateTime.Now);
+            _driverMetrics.TryGetMetric(DriverMetrics.DriverStateMetric, out IMetric stateMetric);
+            ((StringMetric)stateMetric).AssignNewValue(EventPrefix + systemState.ToString());
```

Comment:
this is where I would love to have some stronger typing! if we define our `.TryGetMetric()` method everywhere as
```C#
bool TryGetMetric<T>(string name, out T metric) where T : IMetric
```
then we can write just
```C#
_driverMetrics.TryGetMetric(DriverMetrics.DriverStateMetric, out StringMetric stateMetric);
stateMetric?.AssignNewValue(EventPrefix + systemState);
```
I am sure we can do that.
(also note that the explicit `.ToString()` was redundant)